pkg/util/kube/pod.go

```diff
@@ -110,3 +110,37 @@ func EnsureDeletePod(ctx context.Context, podGetter corev1client.CoreV1Interface
 
 	return nil
 }
+
+// IsPodInAbnormalState checks if the pod is in an abnormal state
+// It could not cover all the cases but we could add more cases in the future
+func IsPodInAbnormalState(pod *corev1api.Pod, log logrus.FieldLogger) bool {
```

Comment:
May need to rename the function to avoid misleading.

IMO, this function checks whether the pod is in an unrecoverable failure state.
Not available but can be recovered is acceptable for this function.

Reply:
Updated